### PR TITLE
Rc/4.0.0

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

Fixes:
- Table
  - allow completely custom elements for TH and TD (pass the same api props to custom els)
  - fixes border striping on table container
  - passes sorting functionality to custom elements
- Footer
  - types & style cleanup
  
**Type of Change**

- [ ] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [ ] No new UIKit components or changes
- [ ] Feature is minimally responsive
- [ ] Manual testing of UI feature
- [ ] Add copyrights to new files
- [ ] Connected ticket to PR

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "@icgc-argo/uikit",
-  "version": "2.5.0",
+  "version": "3.0.1",
```
Comment (justincorrigible) [inline package-lock.json:3]:
```suggestion
  "version": "4.0.0",
```

File: package-lock.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   "lockfileVersion": 2,
   "requires": true,
   "packages": {
     "": {
       "name": "@icgc-argo/uikit",
-      "version": "2.5.0",
+      "version": "3.0.1",
```
Comment (justincorrigible) [inline package-lock.json:9]:
```suggestion
      "version": "4.0.0",
```

**Review by justincorrigible — APPROVED** [review]
LGTM 👍

**Comment by joneubank** [comment]
Echoing Justin's suggestions... the RC branch name and version set don't match up, whats the expected value?

**Review by joneubank — CHANGES_REQUESTED** [review]
Before approving this, can you add a description to the PR with a list of the changes included in this release? The RC PR should list bug fixes, new features, etc. instead of our typical template.

Seeing this list would also help determine if this should be a major/minor/patch version update.

**Comment by ciaranschutte** [comment]
> Before approving this, can you add a description to the PR with a list of the changes included in this release? The RC PR should list bug fixes, new features, etc. instead of our typical template.
> 
> Seeing this list would also help determine if this should be a major/minor/patch version update.

discussed on Slack.
reason for major bump: grabbing the update would require importer to make changes to their code to keep expected functionality.

**Comment by justincorrigible** [comment]
> > Before approving this, can you add a description to the PR with a list of the changes included in this release? The RC PR should list bug fixes, new features, etc. instead of our typical template.
> > Seeing this list would also help determine if this should be a major/minor/patch version update.
> 
> discussed on Slack. reason for major bump: grabbing the update would require importer to make changes to their code to keep expected functionality.

Glad to see that got resolved. Can I request what slack channel this was discussed in?
Ideally, this discussion was documented in the public, topic-focused channels, so the rest of the team are on the same page. I just cannot seem to find which one of those it was.

**Review by joneubank — APPROVED** [review]

**Comment by ciaranschutte** [comment]
> > > Before approving this, can you add a description to the PR with a list of the changes included in this release? The RC PR should list bug fixes, new features, etc. instead of our typical template.
> > > Seeing this list would also help determine if this should be a major/minor/patch version update.
> > 
> > 
> > discussed on Slack. reason for major bump: grabbing the update would require importer to make changes to their code to keep expected functionality.
> 
> Glad to see that got resolved. Can I request what slack channel this was discussed in? Ideally, this discussion was documented in the public, topic-focused channels, so the rest of the team are on the same page. I just cannot seem to find which one of those it was.

sorry it was a PM following up on the PR. Will @ in public channel next time.
Documented the takeaway above.

**Review by justincorrigible — APPROVED** [review]

**Pull request merged**